auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -903,31 +1069,32 @@ private LoadedIdentity parseIdentityContents(final XMLStreamReader streamReader,
                         //modifiable version of Attributes;
                         attributes = new MapAttributes();
                     }
-                    return new LoadedIdentity(name, credentials, attributes, hashEncoding);
+                    return new LoadedIdentity(name, credentials, attributes, hashEncoding, providers);
                 }
-                if (! validNamespace(streamReader.getNamespaceURI())) {
+                if ( !(version.getNamespace().equals(streamReader.getNamespaceURI())) && !(XMLSignature.XMLNS.equals(streamReader.getNamespaceURI())) ) {
```

Comment:
Have you tried parsing an older version of an identity file to make sure that still works?

Reply:
Yeah, it still works with this check